src/switchroot/ostree-prepare-root.c

```diff
@@ -108,63 +168,64 @@ touch_run_ostree (void)
   (void) close (fd);
 }
 
-int
-main(int argc, char *argv[])
+static char*
+resolve_deploy_path (const char * root_mountpoint)
 {
-  const char *root_mountpoint = NULL;
-  char *ostree_target = NULL;
-  char *deploy_path = NULL;
-  char srcpath[PATH_MAX];
   char destpath[PATH_MAX];
-  char newroot[PATH_MAX];
   struct stat stbuf;
-  int orig_cwd_dfd;
-
-  if (argc < 2)
-    {
-      fprintf (stderr, "usage: ostree-prepare-root SYSROOT\n");
-      exit (EXIT_FAILURE);
-    }
-
-  root_mountpoint = argv[1];
+  char *ostree_target, *deploy_path;
 
   ostree_target = parse_ostree_cmdline ();
   if (!ostree_target)
     {
       fprintf (stderr, "No OSTree target; expected ostree=/ostree/boot.N/...\n");
-      exit (EXIT_FAILURE);
-    }
-
-  /* Create a temporary target for our mounts in the initramfs; this will
-   * be moved to the new system root below.
-   */
-  snprintf (newroot, sizeof(newroot), "%s.tmp", root_mountpoint);
-  if (mkdir (newroot, 0755) < 0)
-    {
-      perrorv ("Couldn't create temporary sysroot '%s': ", newroot);
-      exit (EXIT_FAILURE);
+      die ();
     }
 
   snprintf (destpath, sizeof(destpath), "%s/%s", root_mountpoint, ostree_target);
   printf ("Examining %s\n", destpath);
   if (lstat (destpath, &stbuf) < 0)
     {
       perrorv ("Couldn't find specified OSTree root '%s': ", destpath);
-      exit (EXIT_FAILURE);
+      die ();
     }
   if (!S_ISLNK (stbuf.st_mode))
     {
       fprintf (stderr, "OSTree target is not a symbolic link: %s\n", destpath);
-      exit (EXIT_FAILURE);
+      die ();
     }
   deploy_path = realpath (destpath, NULL);
   if (deploy_path == NULL)
     {
       perrorv ("realpath(%s) failed: ", destpath);
-      exit (EXIT_FAILURE);
+      die ();
     }
   printf ("Resolved OSTree target to: %s\n", deploy_path);
-  
+  return deploy_path;
+}
+
+static int
+pivot_root(const char * new_root, const char * put_old)
+{
+  return syscall(__NR_pivot_root, new_root, put_old);
+}
+
+int
+main(int argc, char *argv[])
+{
+  const char *root_mountpoint = NULL;
+  char *deploy_path = NULL;
+  char srcpath[PATH_MAX];
+  struct stat stbuf;
+
+  if (argc < 2)
+    root_mountpoint = "/sysroot";
```

Comment:
> Would it make more sense to have `root_mountpoint = "/"` as the default value, when ostree-prepare-root called without args?

I've remembered why this wasn't a problem for me.  I have a symlink `/sysroot -> .` on my rootfs, so this was being resolved to `/`.